pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -315,7 +315,23 @@ func doCleanSubPath(mounter mount.Interface, fullContainerDirPath, subPathIndex
 	klog.V(4).Infof("Cleaning up subpath mounts for subpath %v", subPathIndex)
 	fullSubPath := filepath.Join(fullContainerDirPath, subPathIndex)
 
-	if err := mount.CleanupMountPoint(fullSubPath, mounter, true); err != nil {
+	quietUnmounter, ok := mounter.(mount.MounterQuietUnmounter)
```

Comment:
Does anything require not-quiet unmount? IMO `umount -q` should be part of regular `CleanupMountPoint`, which can check if the OS supports `umount -q` and call it directly or check if the mount point exists and call plain `umount`.

Question is, how to check for `mount -q` support nicely...

Reply:
The check for whether umount would print 'not mounted' could happen at startup and be returned by kubeVolHost somehow since it is can be though to f as a property of the host.

Reply:
To actually answer your question: in theory all mounts should go thru this codepath to rely on umoutn to check mount table instead of checking mount table themselves, but for this PR i was selfishly fixing only the subpath case. 

Let me try just changing it in `CleanupMountPoint` and see how it goes.

Reply:
related PR https://github.com/kubernetes/kubernetes/pull/109217

Reply:
THX for link. Yes, @manugupt1 this PR is meant to be complementary to https://github.com/kubernetes/kubernetes/pull/109217. for places where we MUST do mount check (the selinux thing), and where the kernel is new, we should be using openat2 for it. But now I am thinking umount is NOT one of those places hence this PR

Reply:
I reworekd the PR to refactor `CleanupMountPoint` and not just `doCleanSubPath`, resolving this thread !